docs/introduction.md

```diff
@@ -0,0 +1,48 @@
+# Introduction
+
+Hera is a Python library that allows you to construct and submit Argo Workflows. It is designed to be intuitive and easy to use, while also providing a powerful interface to the underlying Argo API.
+
+## Hera V5 vs V4
+
+Hera v5 is a major release that introduces breaking changes from v4. The main reason for this is that v5 is a complete rewrite of the library, and is now based on the OpenAPI specification of Argo Workflows. This allows us to provide a more intuitive interface to the Argo API, while also providing full feature parity with Argo Workflows. This means that you can now use all the features of Argo Workflows in your workflows. In addition, the codebase is now much more readable, and the focus can be fully dedicated to improving user workflows.
+
+The library is divided into two main components:
+- `hera.workflows.models` - This package contains the auto-generated code that allows you to construct Argo Workflows. It provides Pydantic models for all the Argo Workflows OpenAPI objects, and allows you to construct workflows using these models. These models are based on the OpenAPI specification, and are therefore exactly the same as the models used by Argo Workflows.
+- `hera.workflows` - This package contains the hand-written code that allows you to construct and submit Argo Workflows. It wraps the auto-generated code, and provides a more intuitive interface to the Argo API. It also provides a number of useful features, such as the ability to submit workflows from a Python function. This package has various extension points that allow you to plug-in the auto-generated models in case you need to use a feature that is not yet supported by the hand-written code.
```

Comment:
```suggestion
- `hera.workflows` - This package contains the hand-written code that allows you to construct and submit Argo Workflows. It wraps the auto-generated code, and provides a more intuitive interface to the Argo API. It also provides a number of useful features, such as the ability to submit workflows from a Python function. This package has various extension points that allow you to plug-in the auto-generated models in case you need to use a feature that is not yet supported by the hand-written code.
- `hera.events.models` - This package contains the auto-generated code that allows you to construct Argo Events. It provides Pydantic models for all the Events objects, and allows you to submit them via the Events service. Note that we only have auto-generated code for Argo Events. In the future, Hera workflows objects will integrate with events.
```